### ISSUE TITLE
Can VS2017 build caffe2 now?

### ISSUE THREAD

**Issue by honamida** [facebookarchive/caffe2#602]
I have built caffe2 successfully in VS2015, but I can't link the library in VS2015 #585 .
I tried to build caffe2 again in VS2017 community, but I got the error message below.

 C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(171,5): error MSB6006: "cmd.exe" 以返回碼 1 結束。 [E:\caffe2\build\caffe2\Caffe2_GPU.vcxproj]

going into GUI and build, I got the link error below 

![image](https://cloud.githubusercontent.com/assets/28566795/26189579/edb79262-3bd7-11e7-998c-fdef5dcf7aed.png)

Also I found that many people asking if  CUDA supports VS2017, and the answer is no from the result after google the key word. Since the installation tutorial of Windows had listed VS17 as an option to compile caffe2, I want to know if there's anyone have built caffe2 in VS17. Thanks for sharing!

**Comment by honamida** [comment]
BTW, if I build without CUDA, it works.

**Comment by Wenchao-Du** [comment]
hi , @honamida , have you build the caffe2 with vs2015 successfully? and i have tried to build caffe2 with vs2015 according to the installation. but it failed, it always showed that "No CMAKE_C_COMPILER could be found.No CMAKE_CXX_COMPILER could be found.", however, i built it with using cmake with param that msvc 14 2015 win64, it could not find it. i have tried many methods, but i have not found a valued way. can you tell me about your successful experiences for building caffe2  with vs2015??? thank you very much!

**Comment by davinwang** [comment]
@Wenchao-Du you can use command
`set CMAKE_GENERATOR="Visual Studio 14 2015 Win64"` to get around the the compiler not found issue, but possibly will encounter other issues.

**Comment by Wenchao-Du** [comment]
@davinwang thanks , i have solved this problem, and i have compiled the caffe2 with CPU X64 successfully, but, i trid to complied the gpu version with cuda 8.0, cudnn 5.1 based on vs2015, but , unluckily, it failed on compiling *.cu file. i  have tried to update the version of cudnn, however it still did not work. do you have some good advice for me ? thank you!

**Comment by davinwang** [comment]
@Wenchao-Du I'm still struggling compiling CPU x64 with VS2015. Did you follow the official guide? Could you share the steps to compile a CPU version? Did you pull caffe2:master?

**Comment by Wenchao-Du** [comment]
@davinwang  yes, i just followed official guide with git clone --recursive ........., and step by step to do, it succeed. but this work have finished two month ago, i am busy on other things. i could share my steps in some days. I suggest you that before you start your work, firstly , you need to update your version of vs2o15, and then install thirdparty python packges. lastly, you could cmake your caffe2-master. good luck.

**Comment by davinwang** [comment]
@Wenchao-Du Thanks. It seems a code issue about `__unused__` was just fixed by Yangqing.

**Comment by davinwang** [comment]
@Wenchao-Du I made some progress in compiling Win+GPU version, firstly you need to checkout tag v0.8.0 rather than master branch because master updates too quickly.
Then you need cherry-pick following commits because they are not included in v0.8.0
4cc5e798941459dc4b26fc1a21b968b3d78434bb
110d0860f1035c83a446e70ba4a37a5f0b44f7ae
d7d0e66ac64767db545d290a06e7758b4479b324
After all, you should still meet some compile error in topk*.cu*, you can comment them out to make the compile work, but I don't know how to fix them gracefully. Hope it helps.

**Comment by Wenchao-Du** [comment]
@davinwang ok, thanks, I will try it soon.  I just want to try it on Ubuntu with GPU, if it works ,I will share with you.

**Comment by davinwang** [comment]
@Wenchao-Du I already switched to Ubuntu+GPU because unstability of Windows build😂
